content/iOS/Practices/iOS-practices.md

```diff
@@ -0,0 +1,118 @@
+
+So the next time I want to do an update, by a glance to the root I know the features my app implements and I can easily navigate to the file I want to update.  
+
+### Files Structure
+
+Within file classes keep related functions together in sections organized with Marks. Some examples for controllers are: View Lifecycle, Layout, User Interaction, Protocol Implementation, etc. 
+
+It is also a nice practice to separate private functions into a private extension, so that a client at a glance at the class can differentiate the interface from the implementation.  
+
+## Architecture Patterns
+
+Be very mindful of the pattern you are gonna follow in your new project and stick with it! 
+
+There's no right answer on what pattern to follow, it depends a lot on your project. I won't be enumerating every option as there are so many, and new ones keep being invented every day according to the new challenges we face as time passes, and we start seeing flaws in the existent ones.
+So, let's stick to the more common ones for now. 
+
+### MVC
+
+Like many others, at first I started coding in MVC, one of the simpler ones. 
+This pattern divides responsibilities into three class types: Models, Views, and Controllers. 
+Models deals with data logic, Views display the UI and the Controllers handles communication between the two, navigation, and business logic.  
+
+The problem with this pattern is that a lot of the responsibilities fall in the controllers' job, making them huge and ugly beasts that can be a real pain to maintain, test, and reuse in the future. 
+
+### MVVM
+
+This pattern divides responsibilities within Models, Views, and View Models, and is one of the most populars.
+Business logic and communication between Models and Views are now responsibility of the ViewModel leaving the Controller responsibility to only handle user interaction, displaying the digested information the View Model provides in Views, and handling navigation. 
+
+This extra separation of concerns allows us to keep a cleaner code base as well as facilitating unit testing.
+
+There are a bunch of variations of this pattern, the most popular ones include a Coordinator that would handle navigation and flow of the app relieving view controllers from this work.  
+
+For an example of this implementation you can check out [Rootstrap's iOS base](https://github.com/rootstrap/ios-base) a very helpful iOS project template to let you jump into your new project quickly. Check this [article](https://www.rootstrap.com/blog/2019/10/25/whats-great-about-our-ios-base/) to find out more about why we did that and how helpful it has been for us.  
+
+## Resource Management 
+
+As we mentioned before we want to have a folder in the root of the project for the resources our app will need, there are many types of resources so its helpful to organize them in different folders according to their type, for example, you may have: property lists, images, gifs, JSONs... and the list goes on.
+
```

Comment:
`As we mentioned before, we want to have a folder in the -root of the project- (project's root?) for the resources our app will need.` I would add a comma after `before`, end the sentence in need and start a new sentence with `There are many...`. And the root of the project or the project's root I'm not sure, just something that came to mind.